crates/js-sys/tests/wasm/RegExp.rs

```diff
@@ -0,0 +1,138 @@
+use wasm_bindgen::JsValue;
+use wasm_bindgen_test::*;
+use js_sys::*;
+
+#[wasm_bindgen_test]
+fn exec() {
+
+    let re = RegExp::new(&JsValue::from("quick\\s(brown).+?(jumps)"), "ig");
+    let result = re.exec("The Quick Brown Fox Jumps Over The Lazy Dog");
+
+    let mut v = vec![];
+    result.unwrap().for_each(&mut |x, _, _| v.push(x));
+
+    assert_eq!(v[0], "Quick Brown Fox Jumps");
+    assert_eq!(v[1], "Brown");
+    assert_eq!(v[2], "Jumps");
+
+    let result = re.exec("foo");
+    assert!(result.is_none());
+}
+
+#[wasm_bindgen_test]
+fn flags() {
+    let re = RegExp::new(&JsValue::from("foo"), "ig");
+    assert_eq!(re.flags(), "gi");
+}
+
+#[wasm_bindgen_test]
+fn global() {
+    let re = RegExp::new(&JsValue::from("foo"), "g");
+    assert!(re.global());
+
+    let re = RegExp::new(&JsValue::from("bar"), "i");
+    assert!(!re.global());
+}
+
+#[wasm_bindgen_test]
+fn ignore_case() {
+    let re = RegExp::new(&JsValue::from("foo"), "");
+    assert!(!re.ignore_case());
+
+    let re = RegExp::new(&JsValue::from("foo"), "i");
+    assert!(re.ignore_case());
+}
+
+#[wasm_bindgen_test]
+fn input() {
+    let re = RegExp::new(&JsValue::from("hi"), "g");
+    re.test("hi there!");
+    assert_eq!(RegExp::input(), "hi there!");
+}
+
+#[wasm_bindgen_test]
+fn last_index() {
+    let re = RegExp::new(&JsValue::from("foo"), "g");
+    re.test("table football, foosball");
+    assert_eq!(re.last_index(), 9);
+
+    let re = RegExp::new(&JsValue::from("foo"), "g");
+    re.set_last_index(10);
+    re.test("table football, foosball");
+    assert_eq!(re.last_index(), 19);
+}
+
+#[wasm_bindgen_test]
+fn last_match() {
+    let re = RegExp::new(&JsValue::from("hi"), "g");
+    re.test("hi there!");
+    assert_eq!(RegExp::last_match(), "hi");
+}
+
+#[wasm_bindgen_test]
+fn last_paren() {
+    let re = RegExp::new(&JsValue::from("(hi)"), "g");
+    re.test("hi there!");
+    assert_eq!(RegExp::last_paren(), "hi");
+}
+
+#[wasm_bindgen_test]
+fn left_context() {
+    let re = RegExp::new(&JsValue::from("world"), "g");
+    re.test("hello world!");
+    assert_eq!(RegExp::left_context(), "hello ");
+}
+
+#[wasm_bindgen_test]
+fn multiline() {
+    let re = RegExp::new(&JsValue::from("foo"), "m");
+    assert!(re.multiline());
+}
+
+#[wasm_bindgen_test]
+fn new() {
+    let re = RegExp::new(&JsValue::from("foo"), "");
+    let re = RegExp::new(&JsValue::from(re), "g");
```

Comment:
This test demonstrates that `RegExp::new` not only can accept a string as its first argument, but also another instance of `RegExp`. But can this be made more ergonomic?

The majority of the use cases of `RegExp` will probably use a string as its first argument, and it would be nice to be able to do `RegExp::new("foo", "g")` instead.

Reply:
One possibility would be to do:

```rust
#[wasm_bindgen]
extern {
    #[wasm_bindgen(constructor)]
    fn new(s: &str, flags: &str) -> RegExp;
    #[wasm_bindgen(constructor)]
    fn new_regexp(r: &RegExp, flags: &str) -> RegExp;
}
```

maybe that could work here?

Reply:
Yes, indeed. I'll update the PR accordingly.